File: scratch/alter_db.js
const mysql = require('mysql2/promise');

async function runMigration() {
  const connection = await mysql.createConnection({
    host: process.env.DB_HOST || process.env.MYSQLHOST || "localhost",
    user: process.env.DB_USER || process.env.MYSQLUSER || "root",
    password: process.env.DB_PASSWORD || process.env.MYSQLPASSWORD || "root",
    database: process.env.DB_NAME || process.env.MYSQLDATABASE || "hospinew",
    port: Number(process.env.DB_PORT || process.env.MYSQLPORT || 3306),
  });

  console.log('Connected to MySQL database.');

  try {
    // 1. user_hospital_bookings
    const [cols1] = await connection.query("SHOW COLUMNS FROM user_hospital_bookings LIKE 'payment_type'");
    if (cols1.length === 0) {
      await connection.query("ALTER TABLE user_hospital_bookings ADD COLUMN payment_type ENUM('Full', 'Part') DEFAULT 'Full' AFTER total_amount");
      await connection.query("ALTER TABLE user_hospital_bookings ADD COLUMN paid_amount DECIMAL(10,2) NULL AFTER payment_type");
      console.log("Added payment_type and paid_amount to user_hospital_bookings");
    } else {
      console.log("user_hospital_bookings already has payment_type");
    }

    // 2. user_lab_test_bookings
    const [cols2] = await connection.query("SHOW COLUMNS FROM user_lab_test_bookings LIKE 'payment_type'");
    if (cols2.length === 0) {
      await connection.query("ALTER TABLE user_lab_test_bookings ADD COLUMN payment_type ENUM('Full', 'Part') DEFAULT 'Full' AFTER total_amount");
      await connection.query("ALTER TABLE user_lab_test_bookings ADD COLUMN paid_amount DECIMAL(10,2) NULL AFTER payment_type");
      console.log("Added payment_type and paid_amount to user_lab_test_bookings");
    } else {
      console.log("user_lab_test_bookings already has payment_type");
    }

    console.log('Migration completed successfully.');
  } catch (err) {
    console.error('Migration error:', err);
  } finally {
    await connection.end();
  }
}

runMigration();

File: scratch/alter_db.py
import mysql.connector
import os

try:
    conn = mysql.connector.connect(
        host="localhost",
        user="root",
        password="",
        database="hospikare"
    )
    cursor = conn.cursor()
    
    # 1. Alter user_hospital_bookings
    try:
        cursor.execute("ALTER TABLE user_hospital_bookings ADD COLUMN payment_type ENUM('Full', 'Part') DEFAULT 'Full' AFTER total_amount")
        print("Added payment_type to user_hospital_bookings")
    except Exception as e:
        print(f"Error adding payment_type to hospital bookings: {e}")
        
    try:
        cursor.execute("ALTER TABLE user_hospital_bookings ADD COLUMN paid_amount DECIMAL(10,2) AFTER payment_type")
        # Update existing records
        cursor.execute("UPDATE user_hospital_bookings SET paid_amount = total_amount WHERE paid_amount IS NULL")
        print("Added paid_amount to user_hospital_bookings")
    except Exception as e:
        print(f"Error adding paid_amount to hospital bookings: {e}")

    # 2. Alter user_lab_test_bookings
    try:
        cursor.execute("ALTER TABLE user_lab_test_bookings ADD COLUMN payment_type ENUM('Full', 'Part') DEFAULT 'Full' AFTER total_amount")
        print("Added payment_type to user_lab_test_bookings")
    except Exception as e:
        print(f"Error adding payment_type to lab bookings: {e}")
        
    try:
        cursor.execute("ALTER TABLE user_lab_test_bookings ADD COLUMN paid_amount DECIMAL(10,2) AFTER payment_type")
        # Update existing records
        cursor.execute("UPDATE user_lab_test_bookings SET paid_amount = total_amount WHERE paid_amount IS NULL")
        print("Added paid_amount to user_lab_test_bookings")
    except Exception as e:
        print(f"Error adding paid_amount to lab bookings: {e}")

    conn.commit()
    cursor.close()
    conn.close()
    print("Database updates completed successfully.")

except Exception as e:
    print(f"Connection error: {e}")
